pennylane/devices/default_clifford.py

```diff
@@ -540,6 +553,8 @@ def simulate(
         global_phase_ops = []
         for op in circuit.operations[use_prep_ops:]:
             gate, wires = _pl_to_stim(op)
+            if isinstance(op, qml.operation.Channel) and not circuit.shots:
+                raise ValueError(f"{self.name} supports error channels only with finite shots.")
```

Comment:
we'd like to hit this at preprocess-time, not execution-time. Can you register this as a pre-processing transform? wondering if we can sneak this into `decompose` somehow... might be possible with another PR happening in tandem that has a custom stopping condition for finite-shot mode. otherwise just a new transform is fine